### PR TITLE
[config] move config.API to api package

### LINKED ISSUE #3722: move config.API into API package
### What would you like to be added
as title

### Why is this needed
part of series refactor effort to break the big `Config` struct into each component's own config

### How important you think this is for IoTeX
- [] must have
- [x] should have
- [] nice to have

### Additional information
Let us know any background or context that would help us better understand the request (for example the particular use-case that prompted this request)

### PR DESCRIPTION
**Pull request by envestcc**
# Description

In order to refactor config.API, the main modifications are:
- move config.GasStation to gasstation package
- move config.API to api package
- refactor unit test of api and gasstation package to remove dependencies on config package

Fixes #3722

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test

**Test Configuration**:
- Firmware version: macOS Monterey 12.5.1
- Hardware: Apple M1 Pro
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by millken — APPROVED** [review]

File: api/serverV2_integrity_test.go [hunk 1]
```diff
@@ -33,14 +33,24 @@ import (
...
+	"github.com/iotexproject/iotex-core/consensus"
 	"github.com/iotexproject/iotex-core/db"
 	"github.com/iotexproject/iotex-core/pkg/unit"
 	"github.com/iotexproject/iotex-core/state/factory"
 	"github.com/iotexproject/iotex-core/test/identityset"
 	"github.com/iotexproject/iotex-core/testutil"
 )
 
+type testConfig struct {
+	API       Config
```
Comment (Liuhaai) [inline api/serverV2_integrity_test.go:45]: Could all fields be private?

**Comment by Liuhaai** [comment]
good job

**Review by Liuhaai — APPROVED** [review]

File: api/serverV2_integrity_test.go (hunk 1 above)
Comment (envestcc) [reply api/serverV2_integrity_test.go]: good suggestion

**Comment by millken** [comment]
should we use `*Config` instead of `Config{}` as parameter, currently the size of the entire `Config{}` has exceeded 2K.

**Comment by envestcc** [comment]
> should we use `*Config` instead of `Config{}` as parameter, currently the size of the entire `Config{}` has exceeded 2K.

Exactly, the `config.Config` is not a small object anymore, using `*config.Config` as parameter can help with performance. specifically, `config.Config` is used as parameter mainly in service initialization and unit test currently, they maybe the major beneficiaries for the proposal.

But should be careful if we want to update the config, because it's shared between several services.

**Comment by CoderZhi** [comment]
> > should we use `*Config` instead of `Config{}` as parameter, currently the size of the entire `Config{}` has exceeded 2K.
> 
> Exactly, the `config.Config` is not a small object anymore, using `*config.Config` as parameter can help with performance. specifically, `config.Config` is used as parameter mainly in service initialization and unit test currently, they maybe the major beneficiaries for the proposal.
> 
> But should be careful if we want to update the config, because it's shared between several services.

good thought, we shouldn't pass `Config` as a whole frequently. Inside of a struct, passing `*` instead of the struct. But let's create a new issue to track it.

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**